src/utilities/TransactionUtilities.ts

```diff
@@ -85,13 +86,15 @@ export const mapTransfer = (transaction: Transaction): CacheTransferTransaction
  * @param {TransferTransaction} transaction
  * @returns {boolean}
  */
-export const cacheDetails = (transaction: TransferTransaction): CACHE => {
-  if (transaction.containsMosaics()) {
-    const cache = transaction.mosaics().find(mosaic => mosaic.mosaicId.namespaceId === CACHE.MOSAICID.namespaceId && mosaic.mosaicId.name === CACHE.MOSAICID.name);
-    if (cache) { return new CACHE(cache.quantity / Math.pow(10, CACHE.DIVISIBILITY)); }
-  }
-  return new CACHE(0);
-};
+export const mosaicDetails = (transaction: TransferTransaction): MosaicTransferable | CACHE => {
+  const mosaics = transaction.mosaics();
+    if (mosaics[0].mosaicId.namespaceId === CACHE.MOSAICID.namespaceId && mosaics[0].mosaicId.name === CACHE.MOSAICID.name) {
+      return new CACHE(mosaics[0].quantity / Math.pow(10, CACHE.DIVISIBILITY));
```

Comment:
We still keeping CACHE ?

Reply:
yeah, why wouldn't we have our token embedded in the sdk?